packages/nextcloud/lib/src/webdav/client.dart

```diff
@@ -75,35 +72,35 @@ class WebDavClient {
       headers: headers,
     );
 
-    request.headers.addAll({
-      ...?rootClient.baseHeaders,
-      ...?rootClient.authentications?.firstOrNull?.headers,
-    });
+    final authentication = rootClient.authentications?.firstOrNull;
+    if (authentication != null) {
+      request.headers.addAll(
+        authentication.headers,
+      );
+    }
 
     // On web we need to send a CSRF token because we also send the cookies.  In theory this should not be required as
     // long as we send the OCS-APIRequest header, but the server has a bug that only triggers when you also send the
     // cookies. On non-web platforms we don't send the cookies so we are fine, but on web the browser always does it
     // and therefore we need this workaround.
     // TODO: Fix this bug in server.
```

Comment:
```suggestion
    // We need to send a CSRF token because we also send the cookies.  In theory this should not be required as
    // long as we send the OCS-APIRequest header, but the server has a bug that only triggers when you also send the cookies.
    // TODO: Fix this bug in server.
```